.github/workflows/publish_integration_docker_image.yaml

```diff
@@ -0,0 +1,84 @@
+name: Publish integration image to DockerHub
+# Builds and publishes graph integration project to DockerHub
+
+# EXAMPLE USAGE: the following is a yaml job statement using this re-usable workflow
+#   publish-image:
+#    uses: jupiterone/.github/.github/workflows/publish_integration_docker_image.yaml@main
```

Comment:
Should we use a major version of this instead of `@main`?

Reply:
Looking at [all the other usages of workflows in this repo](https://github.com/search?p=3&q=org%3AJupiterOne+jupiterone%2F.github%2F.github%2Fworkflows&type=Code), I'm inclined to keep it using main for now. We can make changes to this repo to publish major versions in a subsequent change.

Reply:
@mdaum I'm mainly just concerned that we'll always need to maintain backwards compatibility in this workflow if we recommend `@main`. Maybe we should, but just wanted to throw that out there! 😄

Reply:
I hear that concern and I do think that versioned is better. Was originally thinking we can just address it later, but updating all the usages down the line will be cumbersome. Best to set up tagging for this repo now.

Reply:
https://github.com/JupiterOne/graph-kubernetes/actions/runs/3139289872/jobs/5099547905
was able to change default and confirmed working

Reply:
@austinkelleher I think we can just cut a tag after merging this PR. I think that might be best. Can do so with a commit much like we do in our `graph-*` projects. There is no module to publish, but a tagged commit denoting version should suffice. Thoughts?